src/test/java/org/generationcp/middleware/api/nametype/GermplasmNameTypeServiceImplIntegrationTest.java

```diff
@@ -0,0 +1,49 @@
+package org.generationcp.middleware.api.nametype;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.generationcp.middleware.IntegrationTestBase;
+import org.generationcp.middleware.manager.DaoFactory;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.PageRequest;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+
+public class GermplasmNameTypeServiceImplIntegrationTest extends IntegrationTestBase {
+
+	private DaoFactory daoFactory;
+
+	@Autowired
+	GermplasmNameTypeService germplasmNameTypeService;
+
+	@Before
+	public void setUp() throws Exception {
+		this.daoFactory = new DaoFactory(this.sessionProvder);
+	}
+
+	@Test
+	public void testCreateNameType(){
+		final GermplasmNameTypeRequestDTO germplasmNameTypeRequestDTO = new GermplasmNameTypeRequestDTO();
```

Comment:
move name, code and description to variables, using RandomStringUtils and assert the values after the insertion